tsconfig.json

```diff
@@ -0,0 +1,55 @@
+{
+  "compilerOptions": {
+    "target": "es2017",
+    "allowJs": true,
+    "moduleResolution": "node",
+    "allowSyntheticDefaultImports": true,
+    "noImplicitAny": true,
```

Comment:
I always approach a large TS conversion by going through the following process:

1) Making a first pass w/ `"noImplicitAny": false` across the whole codebase. The main goal is to convert each file to `.ts`. This involves dealing with the subset of JS that TS is unhappy with (i.e., ES6 classes that don't state their member field types up front, extra stuff pinned onto functions, etc...)
2) Set  `"noImplicitAny": true` and make all the `any`s explicit instead of implicit
3) Get rid of any remaining `any`s as possible

Starting out with strict TS config will result in a lot of thrash, as new type-checking issues surface in waves (i.e., a commit that improves types in file B reveals type-checking failures in file A -- over and over and over)